server/src/actors/outcome.rs

```diff
@@ -0,0 +1,312 @@
+//! This module contains the actor that forwards Outcomes to a kafka topic
+//!
+
+use std::sync::Arc;
+use std::time::Instant;
+
+use actix::prelude::*;
+use chrono::{DateTime, SecondsFormat, Utc};
+use serde::{Serialize, Serializer};
+
+use semaphore_common::Config;
+use semaphore_general::protocol::EventId;
+use semaphore_general::reason::OutcomeReason;
+
+use crate::utils::instant_to_system_time;
+use crate::ServerError;
+
+/// Message
+#[derive(Debug, Serialize)]
+pub struct KafkaOutcomeMessage {
+    /// The timespan of the event outcome.
+    #[serde(serialize_with = "instant_serializer")]
+    pub timestamp: Instant,
+    /// Organization id.
+    pub org_id: Option<u64>,
+    /// Project id.
+    pub project_id: Option<u64>,
+    /// The DSN project key id.
+    pub key_id: Option<u64>,
+    /// The outcome.
+    pub outcome: Outcome,
+    /// Reason for the outcome.
+    pub reason: OutcomeReason,
+    /// The event id.
+    pub event_id: Option<EventId>,
+}
+
+impl Message for KafkaOutcomeMessage {
+    type Result = Result<(), OutcomeError>;
+}
+
+/// Serialization function for Instant
+/// Turns an instant in a RFC 3339 formatted date with the shape YYYY-MM-DDTHH:MM:SS.mmmmmmZ
+/// e.g. something like: "2019-09-29T09:46:40.123456Z"
+fn instant_serializer<S>(time: &Instant, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let start_time = instant_to_system_time(&time);
+    let date_time: DateTime<Utc> = start_time.into();
+    let formatted_str = date_time.to_rfc3339_opts(SecondsFormat::Micros, true);
+    serializer.serialize_str(formatted_str.as_str())
+}
+
+/// Defines the possible outcomes from processing an event
+#[derive(Debug, Clone, Copy)]
+#[repr(u8)]
+pub enum Outcome {
+    #[allow(dead_code)]
+    Accepted = 0,
+    #[allow(dead_code)]
+    Filtered = 1,
+    RateLimited = 2,
+    Invalid = 3,
+    #[allow(dead_code)]
+    Abuse = 4,
+}
+
+impl Serialize for Outcome {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        serializer.serialize_u8(*self as u8)
+    }
+}
+
+impl Outcome {
+    /// Returns the name of the outcome as recognized by Sentry.
+    pub fn to_name(&self) -> &'static str {
+        match self {
+            Outcome::Accepted => "accepted",
+            Outcome::Filtered => "filtered",
+            Outcome::RateLimited => "rate_limited",
+            Outcome::Invalid => "invalid",
+            Outcome::Abuse => "abuse",
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn outcome_serialization_test() {
+        let val = serde_json::to_string(&Outcome::Accepted).unwrap_or("error".to_string());
```

Comment:
What does this test do?

Reply:
Nothing, used to see how a serialised message looks, left there by mistake. Thanks for noticing it.